### ISSUE TITLE
Win11 Py3.11 install workaround

### ISSUE THREAD

**Issue by opcris** [pybluez/pybluez#471]
*If you are trying to install 'pybluez' using pip under Win11 and still get the exit with code 1 (error originated from a subprocess) try this:*

## command prompt:
> python -m pip install -e git+https://github.com/pybluez/pybluez.git#egg=pybluez

**Comment by LucasPinheiro33** [comment]
Hello, How extactly i write this in the prompt, because i write this and give a error like the image.
![pybluez](https://user-images.githubusercontent.com/128330249/226215933-0c8ca9bc-1fd3-4d64-a29c-c9debdae5030.png)
I appreciate any help.

**Comment by opcris** [comment]
Hi, Lucas,
Apparently, you neither have git installed nor git is found in your PATH.
To install git in Windows go to the git official download page and install it from there.
To add it to PATH go to 'System properties' -> 'Environment variables' and edit the 'Path' in 'System variables' -> paste the path to your git/cmd folder.
I hope this will help, good luck!

**Comment by borispopov** [comment]
Hi @opcris, I tried using that prompt to install pybluez but I still get exit code: 1. I tried on python 3.10.11 and 3.11.3 but neither work, I am on windows 11. Error code below:
<img width="704" alt="image" src="https://user-images.githubusercontent.com/22267292/230980348-a6770f1e-03ef-4a93-8bb8-92cb66a54495.png">
Thanks in advance

**Comment by opcris** [comment]
Hello, @borispopov 
exit code: 1 may mean 'operation not permited'?
try install / upgrade pip, setuptools, ez_setup (see if installed and which version with 'pip list')
pip install --upgrade setuptools
or install setuptools if not installed
Let me know if it worked.
Good luck!
C.

**Comment by borispopov** [comment]
Thanks for the reply!
pip and setuptools were up to date and I installed ez_setup. The current versions of each are below:
pip 23.0.1
setuptools 67.6.1
ez_setup 0.9
I'm still getting the same error.
Note: I am using python 3.10

**Comment by opcris** [comment]
Hello @borispopov,
Sorry about that. I ran out of ideas :) The typical solution to this problem is to check the Python version and those tools plus the required Windows dependencies. The latest versions should work together.
The[ official installation instructions ](https://github.com/pybluez/pybluez/blob/master/docs/install.rst)page.

**Comment by mpariente** [comment]
@borispopov, you should show the entire error message, specially the bottom part which shows the root cause.

**Comment by CrazyBrain999** [comment]
@borispopov  instalar copilador c++ en la pagina de Windows, y despues intalar libreria bajo el siguiente comando:
pip install git+https://github.com/pybluez/pybluez.git#egg=pybluez

**Comment by Vypersan** [comment]
@opcris  Holy shit thank you, I have been struggling with it for the past two months. I was very close to just downgrading to win10 again

**Comment by opcris** [comment]
> @opcris Holy shit thank you, I have been struggling with it for the past two months. I was very close to just downgrading to win10 again

I feel you :)) Cheers!

**Comment by AmberElferink** [comment]
Thanks for posting, that got me further but I now get this issue when I try this: 
```
 LINK : fatal error LNK1158: cannot run 'rc.exe'
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\link.exe' failed with exit code 1158
 ```
 Any idea what could be the matter?

**Comment by opcris** [comment]
>  LINK : fatal error LNK1158: cannot run 'rc.exe'
>     error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\link.exe' failed with exit code 1158

Hi Amber, it looks like the linker (link.exe) is unable to find or execute the resource compiler (rc.exe). First, make sure that Visual Studio is installed correctly on your system (specifically, check the "Path" variable and ensure that it includes the correct path to the Visual Studio installation directory) and ensure that the necessary paths are set correctly in your system's environment variables.

**Comment by franronci** [comment]
Can you solved the problem?

**Comment by franronci** [comment]
could you solved the problem?

**Comment by AmberElferink** [comment]
In the end I did not put in more effort, and just fixed it on Linux, where I needed it most

**Comment by MergingMonkey** [comment]
I don't think it should work and it feels like a hacky workaround, but it did it for me:
Copy `rc.exe` and `rcdll.dll` from `C:\Program Files (x86)\Windows Kits\10\bin\<highest version>\x64` to `C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin`.
Taken from [here](https://www.tutorialexample.com/fix-python-pip-install-link-fatal-error-lnk1158-cannot-run-rc-exe-error-python-tutorial/)
